ibis/expr/analysis.py

```diff
@@ -935,9 +935,24 @@ def validate(self, expr):
     def _among_roots(self, node):
         return self.roots_shared(node) > 0
 
+    def compatible_with(self, a, b):
+        # self and other are equivalent except for predicates, selections, or
+        # sort keys any of which is allowed to be empty. If both are not empty
+        # then they must be equal
+        if a.equals(b):
+            return True
```

Comment:
It looks like this branch is never hit. Should we remove this and the previous line?